Add ADR on commit structure

gbirke opened the pull request.

gbirke committed:
Add ADR on commit structure

Abban commented on docs/adr/005-git-commit-structure.md:
```diff
@@ -0,0 +1,143 @@
+
+## Decision
+
+### Option 1: Get rid of "current banner" concept
```
We already point the config file to the banner entrypoints, so having a structure like `banners/desktop/C24_blah_01, banners/desktop/C24_blah_02` would be an easy enough change.

It would also remove the archive step (and the archive), the prepare campaign step would be duplicating and renaming the banner folder for the next test. Way less merge issues because different banner branches would be working on code in different places so not confusing git so much.

The downside would be a lot of folders in the busy banner channels towards the end of the campaign, but I can deal with that. I'd prefer it over having to update a symlink every test.

moiikana replied:
I also like the clear approach of having just straight up "ready to work" folders for all the banners, 
even though I fear the same as Abban -> more mistakes, accidental changes in other folders. 
Would be so cool if you could _lock_ folders locally in your IDE like with layers in some painting software.

Abban committed:
Add new banner commit structure to ADR

Sperling-0 commented on docs/adr/005-git-commit-structure.md:
```diff
@@ -0,0 +1,162 @@
+We can now merge banner feature branches in any order, without conflicts
+in the banner code. The only conflict that can occur is in
+`campaign_info.toml`, but we'll resolve that conflict with the following
+rule: Always merge the version that has the higher test number.
```
"Always merge the version that has the higher test number."
What is the meaning of _higher test number_ here?

gbirke replied:
I'll add an explanation and example

gbirke replied:
I've added another commit with example

gbirke committed:
Explain merge strategy and test structure

We discussed the banner test behavior and decided to go with a
CHANNELNAME approach until https://phabricator.wikimedia.org/T369276 is
done.

Added a clarifying section to explain the merge strategy.
